Core/GraphicsInterops/Source/ExternalTextureImpl.h

```diff
@@ -0,0 +1,35 @@
+#include <Babylon/Graphics/Device.h>
+#include <Babylon/GraphicsInterops/ExternalTexture.h>
+#include <Babylon/Graphics/RendererType.h>
+#include <Babylon/Graphics/Texture.h>
+#include <Babylon/Graphics/DeviceContext.h>
+
+#include <napi/env.h>
+#include <napi/napi_pointer.h>
+
+namespace Babylon::Graphics
+{
+    struct ExternalTexture::Impl
+    {
+        Impl(TextureType nativeTexture);
+        ~Impl();
+
+        //TODO: Change this to take a context.
+        Napi::Promise AddToContext(Napi::Env& env);
+
+        uint32_t GetWight();
+
+        uint32_t GetHeight();
+
+        uint32_t GetFormat();
+
+    private:
+        bgfx::TextureFormat::Enum m_format{};
+        uintptr_t m_nativeTexture{};
+        uint32_t m_width{};
+        uint32_t m_height{};
+        uint32_t m_mips{};
+
+        void GetNativeTextureData(TextureType nativeTexture);
```

Comment:
nit: Having a void-returning function called "Get" is a little confusing, took me a moment to figure out what it was doing. `CacheNativeTextureData(...)`, maybe? Or `ReadPropertiesFromNativeTexture(...)`, or something like that?

Reply:
Alternatively, if the only reason this is a member is so that it can set the local property caches, it might also be possible to have it return a struct (or even `auto`) containing the values rather than being a member. That way the name `GetNativeTextureData(...)` fits because it's fetching and returning the native texture data, the "ExternalTexturePlatform.cpp"s could become headers instead, and `ExternalTexture::Impl` would just call into them to get the values out. In fact, I think that would remove the reason to separate out `ExternalTexture::Impl` into its own file with its own header; it could go back to being an implementation detail inside "ExternalTexture.cpp," unless there's another reason that I'm missing to have it externalized.

Reply:
I like renaming it to ReadPropertiesFromNativeTexture, I don't like creating a new struct just for storing this values temporarily. I also removed ```ExternalTexture::Impl``` completely and moved the platform specific code to match the way it is been handled by Graphics, I think it is better organized now.